lib/SYCL/SYCLArgsFlattening.cpp

```diff
@@ -566,36 +602,49 @@ static bool isSafeToPromoteArgument(Argument *Arg, bool isByValOrInAlloca,
         // TODO: This runs the above loop over and over again for dead GEPs
         // Couldn't we just do increment the UI iterator earlier and erase the
         // use?
+	DEBUG(dbgs() << "SYCL: Dead GEP" << *GEP << "\n");
         return isSafeToPromoteArgument(Arg, isByValOrInAlloca, AAR,
                                        MaxElements);
       }
 
       // Ensure that all of the indices are constants.
       for (User::op_iterator i = GEP->idx_begin(), e = GEP->idx_end();
-        i != e; ++i)
-        if (ConstantInt *C = dyn_cast<ConstantInt>(*i))
+        i != e; ++i) {
+	DEBUG(dbgs() << "SYCL: " << Arg->getName() << " used in GEP: " << *GEP << "\n");
+        if (ConstantInt *C = dyn_cast<ConstantInt>(*i)) {
+	  DEBUG(dbgs() << C->getSExtValue() << " constant extend value.\n");
           Operands.push_back(C->getSExtValue());
-        else
+        } else {
+	  DEBUG(dbgs() << "Not a constant operand GEP.\n");
           return false;  // Not a constant operand GEP!
+        }
+      }
 
       // Ensure that the only users of the GEP are load instructions.
       for (User *GEPU : GEP->users())
         if (LoadInst *LI = dyn_cast<LoadInst>(GEPU)) {
           // Don't hack volatile/atomic loads
-          if (!LI->isSimple()) return false;
+          if (!LI->isSimple()) return false; 
           Loads.push_back(LI);
         } else {
           // Other uses than load?
+	  DEBUG(dbgs() << "SYCL: " << Arg->getName() << " used in GEP: " << *GEP << " in " <<  F->getName() << "\n");
+	  DEBUG(dbgs() << "Has other uses than load.\n");
```

Comment:
2 indentation errors.

Reply:
Furthermore, you can fuse the message output with only 1 `DEBUG()` everywhere in the following.